### PR TITLE
don't do pod deletion management for pipeline builds

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@csrwng ptal

**Comment by bparees** [comment]
@csrwng bump

**Comment by csrwng** [comment]
@bparees sorry I missed this one earlier. What does it mean when your pipeline build is still running but a pod associated with the build is deleted?

File: pkg/build/controller/controller.go [hunk 1]
```diff
@@ -373,9 +373,13 @@ func (bc *BuildPodDeleteController) HandleBuildPodDeletion(pod *kapi.Pod) error
 	}
 	build := obj.(*buildapi.Build)
 
+	if build.Spec.Strategy.JenkinsPipelineStrategy != nil {
```
Comment (csrwng) [inline pkg/build/controller/controller.go]: ut?

Comment (bparees) [inline pkg/build/controller/controller.go]: ut?

Comment (csrwng) [inline pkg/build/controller/controller.go]: unit test :)

**Comment by bparees** [comment]
>  sorry I missed this one earlier. What does it mean when your pipeline build is still running but a pod associated with the build is deleted?

it means craziness happened because there should never be a pod associated w/ a pipeline build.

but because of some of the weird things we do to handle resync scenarios, it's possible a "pod deletion" event could get generated for a pod that never existed, which could cause us to mark the pipeline build as "ERROR/FAIL" (which is what we normally do when a pod disappears for a running build).

so i'm just trying to make very sure we don't do anything with pipeline builds as part of the pod deletion handling logic.  (and conversely, if the pipeline build is deleted, that we don't accidentally terminate an innocent pod)

**Comment by csrwng** [comment]
got it

File: pkg/build/controller/controller.go (hunk 1 above)
Comment (bparees) [inline pkg/build/controller/controller.go]: you and your high standards.  yeah let me do that.

**Comment by bparees** [comment]
@csrwng UTed.

**Comment by csrwng** [comment]
LGTM

**Comment by bparees** [comment]
[merge]

Ben Parees | OpenShift

**Pull request merged**